src/components/AddRepositoryDetails/AddRepo.tsx

```diff
@@ -162,17 +230,51 @@ export default function AddRepo() {
             jsonCompute.map((e:any) => {
                 js.push({id:e.id, name:e.service_name});
             })
+            
             setComputeServices(js);
-        }
+            //fetchComputeService();
 
+        }
     }
 
     const handleComputeSelect = (e:any) => {
-        console.log("select");
+        console.log(e.target);
+        if (e.target.value== ""){
+            setSelectComputeService("");
+        }
+        else{
+            setSelectComputeService(e.target.value);
+        }
+    }
+
+
+    const handleConfigure = async () => {
+        console.log("configure");
+
+                
+        if (singleCheckedData.name=="" || singleCheckedData.link=="" || singleCheckedData.org=="" || selectComputeService==""){
+            console.log("One or many fields are empty");
+        }
+        else{
+            const requestOptions = {
+                method: "POST",
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({
+                    repo_name: singleCheckedData.name,
+                    repo_link: singleCheckedData.link,
+                    login_name: singleCheckedData.org,
+                    service_name: selectComputeService
+                })
+            };
+            const responseCompute = await fetch(`http://localhost:5001/api/config_repos`, requestOptions);
```

Comment:
I'd like to try to keep the logic for interacting with the backend services outside of the react components, so that it's all in one place and can be refactored more easily in the future. could you move this into a new method `configureRepository` in the `DashboardController` class, and to use the base url that is stored there instead of hardcoded to localhost:5001?

Reply:
Updated